addons/sale/models/product_product.py

```diff
@@ -56,6 +56,12 @@ def _get_combination_info_variant(self, add_qty=1, pricelist=False, parent_combi
         self.ensure_one()
         return self.product_tmpl_id._get_combination_info(self.product_template_attribute_value_ids, self.id, add_qty, pricelist, parent_combination)
 
+    def _filter_to_unlink(self):
+        domain = [('product_id', 'in', self.ids)]
+        lines = self.env['sale.order.line'].read_group(domain, ['product_id'], ['product_id'])
+        linked_products = [group['product_id'][0] for group in lines]
+        return super(ProductProduct, self - self.browse(linked_products))._filter_to_unlink()
```

Comment:
Variable identifiers are a bit misleading.  I suggest to either use `linked_product_ids` or
```suggestion
        linked_products = self.browse([group['product_id'][0] for group in lines])
        return super(ProductProduct, self - linked_products)._filter_to_unlink()
```